### DISCUSSION TITLE
Add Edifact Mapping

### DISCUSSION THREAD

**Discussion by Yertuad** [php-edifact/edifact#116] [Q&A]
Hello there.

I wish to use edifact and edifact-mapping to parse and interpret a file with D93A specification.
Unfortunately this mapping dosen't exist on edifact-mapping bundle.

My question is : where did you find or how did you create segment.xml and messages xml files for other specifiations ? Is it possible in any way to generate the mapping for D93A ? 

Thank's a lot in advance,
Have a good day.

**Comment by sabas** [comment]
Hi, the issue with older directories is that the original data for the specification doesn't really follow the newer editions format so I didn't parse them correctly...
You can have a look here to see how I did
https://github.com/php-edifact/untdid-parser

If you don't want to tackle the automatic conversion you could use the D94A and compare to the unece untdid...

**Reply by Yertuad** [reply]
Thank's a lot for your quick answer.

Unfortunately I can't make the runner.php to work on the downloaded D93A zip from unece.org.
I also tried my own way by calling the EDSD, UNCL and EDED Parser but it seems that the regex are not working. I always get an "undefined offset" error each time I try. It's always due to a regex not working. Modifying the regex looks to be working on the EDSDParser.php but it only gives me a segment.xml.

I am not very used to the regex syntax and even less to UN/EDIFACT process and how it really works. Therefore I think I will just use the READER to parse my files and then do my custom treatments on the multidimensional array I get to categorize the data I need.

If you have any further informations or help in order to include this D93A mapping to your php-edifact mapping project don't hesitate to tell me. 

Anyway thank you a lot for your work and tour answer. 

Have a good day.

**Reply by sabas** [reply]
Try to copy and rename this to match D93A and make changes as needed 😁

https://github.com/php-edifact/edifact-mapping/tree/master/src/Mapping/D94B
